test/unit/math/rev/arr/functor/integrate_ode_rk45_test.cpp

```diff
@@ -116,3 +116,90 @@ TEST(StanAgradRevOde_integrate_ode_rk45, lorenz_finite_diff) {
 
   test_ode(lorenz, t0, ts, y0, theta, x, x_int, 1e-8, 1e-1);
 }
+
+TEST(StanAgradRevOde_integrate_ode_rk45, rate_as_param) {
+  using stan::math::integrate_ode_rk45;
+  using stan::math::to_var;
+
+  const double t0 = 0.0;
+  harm_osc_ode_fun ode;
+  std::vector<double> theta{0.15};
+  std::vector<double> y0{1.0, 0.0};
+  std::vector<stan::math::var> ts;
+  for (int i = 0; i < 100; i++)
+    ts.push_back(t0 + 0.1 * (i + 1));
+  std::vector<double> x;
+  std::vector<int> x_int;
+  std::vector<stan::math::var> y0v = to_var(y0);
+  std::vector<stan::math::var> thetav = to_var(theta);
+
+  std::vector<std::vector<stan::math::var> > res;
+  auto test_val = [&res]() {
+    EXPECT_NEAR(0.995029, res[0][0].val(), 1e-5);
```

Comment:
I'm using the solver with default tolerance(1e-6), so I'm expecting the error to be of same scale, which is true according to the numerical results. The tolerance is bit greater because to "the same scale" does necessarily "less than". The test tolerance could be lowered by solving the ODEs with lower tolerance.